### ISSUE TITLE
Add caching to /statistics endpoint

### ISSUE THREAD

**Issue by Erb3** [modrinth/code#2544]
This endpoint is significantly slower than a lot of others, due to doing large database queries. Caching this endpoint would improve the response times, and reduce DDoS risk.

**Comment by Erb3** [comment]
I had an open PR on the labrinth repo for this, however I have decided to not re-create it in the monorepo. This is primarily because the code was too repetitive, and I think there's likely a better way to do it. I still think it's a valid issue however.

**Comment by IMB11** [comment]
Closing due to #5897 , @tdgao should consider tanstack caching on the frontend for requests in a close timespan with the same queries.

**Comment by piprett** [comment]
@IMB11 I believe this issue has not been solved, and will not be solved in the linked PR. The issue is about the global labrinth statistics endpoint, not analytics, not frontend. See [the comment](https://github.com/modrinth/labrinth/pull/949#issuecomment-2285820622) you made in the old labrinth PR.

The performance looks to have become somewhat better (infra upgrades?), but still worth server-side caching imo.

```bash
$ time curl https://api.modrinth.com/v3/statistics
{"projects":123591,"versions":1104527,"authors":53041,"files":1205368}
real	0m2.035s
user	0m0.042s
sys		0m0.009s
```

**Comment by Prospector** [comment]
i don't think this is a significant issue and likely some sort of caching happens in redis? unsure. anyway, if an issue our infra engineer will bring it up
